pkg/streamingpromql/operator/instant_vector_selector.go

```diff
@@ -48,9 +47,7 @@ func (v *InstantVectorSelector) NextSeries(_ context.Context) (InstantVectorSeri
 
 	v.memoizedIterator.Reset(v.chunkIterator)
 
-	data := InstantVectorSeriesData{
-		Floats: GetFPointSlice(v.numSteps), // TODO: only allocate this if we have any floats (once we support native histograms)
-	}
+	data := InstantVectorSeriesData{}
 
 	for stepT := v.Selector.Start; stepT <= v.Selector.End; stepT += v.Selector.Interval {
 		var t int64
```

Comment:
[nit] What do you think about renaming `val` to `f` here? Would make things a little more symmetrical with `t` and `h`.